src/Admin/SiteHealth.php

```diff
@@ -171,6 +197,116 @@ public function persistent_object_cache() {
 		];
 	}
 
+	/**
+	 * Get the test result data for whether there is page cache or not.
+	 *
+	 * @return array
+	 */
+	public function page_cache() {
+
+		$is_using_page_cache = $this->is_site_has_page_cache();
+
+		$badge_color = 'red';
+		$status      = 'critical';
+		$label       = __( 'Page caching is not enabled.', 'amp' );
+
+		if ( true === $is_using_page_cache['server_caching'] && false === $is_using_page_cache['client_caching'] ) {
+			$badge_color = 'orange';
+			$status      = 'recommended';
+			$label       = __( 'Page caching is enabled, but client caching headers are missing.', 'amp' );
+		} elseif ( true === $is_using_page_cache['server_caching'] && true === $is_using_page_cache['client_caching'] ) {
+			$badge_color = 'green';
+			$status      = 'good';
+			$label       = __( 'Page caching is enabled.', 'amp' );
+		}
+
+		return [
+			'badge'       => [
+				'label' => $this->get_badge_label(),
+				'color' => $badge_color,
+			],
+			'description' => esc_html__( 'The AMP plugin performs at its best when page caching is enabled.', 'amp' ),
+			'test'        => 'amp_page_cache',
+			'status'      => $status,
+			'label'       => esc_html( $label ),
+		];
+	}
+
+	/**
+	 * Check if site has page cache enable or not.
+	 *
+	 * @return array array(
+	 *                  server_caching: True if site has page cache. Otherwise False.
+	 *                  client_caching: True if site has page cache and passed appropriate header.
+	 *               )
+	 */
+	private function is_site_has_page_cache() {
+
+		$has_page_cache = [
+			'server_caching' => false,
+			'client_caching' => false,
+		];
+		$response_list  = [];
+		$request_args   = [
+			'headers' => [
+				'X-AMP-Request-Random-Number' => true,
+				'Cache-Control'               => 'max-age=' . ( YEAR_IN_SECONDS + MONTH_IN_SECONDS ),
+			],
+		];
+
+		for ( $i = 1; $i <= 3; $i++ ) {
+			$http_response        = wp_remote_get( home_url(), $request_args );
+			$http_response_header = wp_remote_retrieve_headers( $http_response );
+			$http_response_header = ( ! empty( $http_response_header ) ) ? $http_response_header : [];
+			$http_response_header = ( is_array( $http_response_header ) ) ? $http_response_header : $http_response_header->getAll();
+
+			$response = false;
+
+			// If any of header tag exist. and x-amp-random-number doesn't consider page cache exists.
+			if (
+				! empty( $http_response_header['cache-control'] ) ||
+				! empty( $http_response_header['expires'] ) ||
+				! empty( $http_response_header['last-modified'] ) ||
+				! empty( $http_response_header['etag'] )
+			) {
+				$has_page_cache['client_caching'] = true;
+			}
+
+			if ( ! empty( $http_response_header['x-amp-random-number'] ) ) {
+				$response = $http_response_header['x-amp-random-number'];
+			} elseif ( empty( $http_response_header['x-amp-random-number'] ) ) {
```

Comment:
Condition is always `true` because `! empty( $http_response_header['x-amp-random-number'] )` is already `false` at this point.

```suggestion
			} else {
```